index.html

```diff
@@ -312,182 +2212,119 @@ <h2>
+        at <a href=
+        "https://github.com/WICG/webhid/blob/main/blocklist.txt">https://github.com/WICG/webhid/blob/main/blocklist.txt</a>
+      </p>
+      <p>
+        A report is a <dfn>blocked report</dfn> if the following steps return
+        `true`:
       </p>
       <ol>
-        <li>If |filter|.{{HIDDeviceFilter/vendorId}} is present and
-        |device|.{{HIDDevice/vendorId}} does not equal
-        |filter|.{{HIDDeviceFilter/vendorId}}, return `mismatch`.
-        </li>
-        <li>If |filter|.{{HIDDeviceFilter/productId}} is present and
-        |device|.{{HIDDevice/productId}} does not equal
-        |filter|.{{HIDDeviceFilter/productId}}, return `mismatch`.
+        <li>Repeat for each |rule| in the [=HID blocklist=]:
+          <ol>
+            <li>If the report is [=blocked by a blocklist rule=] |rule| then
+            return `true`.
+            </li>
+          </ol>
         </li>
-        <li>If |filter|.{{HIDDeviceFilter/usagePage}} is present, iterate over
-        the {{HIDCollectionInfo}} collections in
-        |device|.{{HIDDevice/collections}}. If there is no [=matching
-        collection=], return `mismatch`.
-        </li>
-        <li>Return `match`.
+        <li>Return `false`.
         </li>
       </ol>
       <p>
-        A {{HIDCollectionInfo}} |collection| is a <dfn>matching
-        collection</dfn> for a {{HIDDeviceFilter}} |filter| if the following
-        steps return match:
+        A report is <dfn>blocked by a blocklist rule</dfn> |rule| if the
+        following steps return `true`:
       </p>
       <ol>
-        <li>If |filter|.{{HIDDeviceFilter/usagePage}} is present and
-        |collection|.{{HIDCollectionInfo/usagePage}} does not equal
-        |filter|.{{HIDDeviceFilter/usagePage}}, return mismatch.
+        <li>Let |reportId:octet| be the report ID of the report, or 0 if the
+        device does not use report IDs.
         </li>
-        <li>If |filter|.{{HIDDeviceFilter/usage}} is present and
-        |collection|.{{HIDCollectionInfo/usage}} does not equal
-        |filter|.{{HIDDeviceFilter/usage}}, return mismatch.
+        <li>Let |reportType| be `"input"` if the report is an input report,
+        `"output"` if the report is an output report, or `"feature"` if the
+        report is a feature report.
         </li>
-        <li>Return match.
+        <li>Let |collection:HIDCollectionInfo| be the [=top-level collection=]
+        that contains the report.
         </li>
-      </ol>
-      <p>
-        A {{HIDDeviceFilter}} |filter| <dfn>is valid</dfn> if the following
-        steps return valid:
-      </p>
-      <ol>
-        <li>If |filter|.{{HIDDeviceFilter/productId}} is present and
-        |filter|.{{HIDDeviceFilter/vendorId}} is not present, return invalid.
+        <li>If |rule| contains `vendor` and |rule|.`vendor` is not
+        |device|.{{HIDDevice/vendorId}}, return `false`.
+        </li>
+        <li>If |rule| contains `product` and |rule|.`product` is not
```

Comment:
Can a rule contain a product ID without a vendor ID?

Reply:
It's allowed by the spec but it's probably not a good idea to have rules that filter on a product ID without specifying a vendor ID, or a usage ID without specifying a usage page.

Since we control the content of the blocklist I felt it wasn't important to restrict rule definitions in the spec itself.